src/EditorFeatures/Next/Options/EditorConfigDocumentOptionsProvider.cs

```diff
@@ -104,15 +110,34 @@ private void Workspace_DocumentOpened(object sender, DocumentEventArgs e)
                 // We don't have anything cached, so we'll just get it now lazily and not hold onto it. The workspace layer will ensure
                 // that we maintain snapshot rules for the document options. We'll also run it on the thread pool
                 // as in some builds the ICodingConventionsManager captures the thread pool.
-                var conventionsAsync = Task.Run(() => _codingConventionsManager.GetConventionContextAsync(path, cancellationToken));
+                var conventionsAsync = Task.Run(() => GetConventionContextAsync(path, cancellationToken));
+                var context = await conventionsAsync.ConfigureAwait(false);
+                if (context == null)
+                {
+                    // exception thrown while trying to read the editorconfig file
+                    return null;
+                }
 
-                using (var context = await conventionsAsync.ConfigureAwait(false))
+                using (context)
                 {
                     return new DocumentOptions(context.CurrentConventions);
                 }
             }
         }
 
+        private async Task<ICodingConventionContext> GetConventionContextAsync(string path, CancellationToken cancellationToken)
+        {
+            try
+            {
+                return await _codingConventionsManager.GetConventionContextAsync(path, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception)
+            {
+                // exception may be thrown while attempting to read editor config file
+                return null;
```

Comment:
Rather than returning null can we return some sort of "empty" construct so we don't need the checks everywhere else?